core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -2135,23 +2135,23 @@ public void _testLikeAndSimilarFails() {
     expr("INTERVAL '-5-4' YEAR TO MONTH").assertInterval(is(-64L));
   }
 
-  @Test void testIntervalMillisConversion() {
-    expr("INTERVAL '1' DAY").assertInterval(is(86_400_000L));
-    expr("INTERVAL '1' HOUR").assertInterval(is(3_600_000L));
-    expr("INTERVAL '1' MINUTE").assertInterval(is(60_000L));
-    expr("INTERVAL '1' SECOND").assertInterval(is(1_000L));
-    expr("INTERVAL '1:05' HOUR TO MINUTE").assertInterval(is(3_900_000L));
-    expr("INTERVAL '1:05' MINUTE TO SECOND").assertInterval(is(65_000L));
-    expr("INTERVAL '1 1' DAY TO HOUR").assertInterval(is(90_000_000L));
-    expr("INTERVAL '1 1:05' DAY TO MINUTE").assertInterval(is(90_300_000L));
-    expr("INTERVAL '1 1:05:03' DAY TO SECOND").assertInterval(is(90_303_000L));
+  @Test void testIntervalNanosConversion() {
```

Comment:
The base unit is changed to nanosecond, so I modified/deleted a lot of tests here

Reply:
This area is a red flag. If you have to change this much code in a fork, we're likely using the fork improperly and potentially changing behavior we don't mean to change. We should tread very carefully here.